modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalResponse.java

```diff
@@ -88,8 +88,8 @@ public String toString() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeDouble(metricScore);
-        out.writeMap(details, StreamOutput::writeString, (o, v) -> v.writeTo(o));
-        out.writeMap(failures, StreamOutput::writeString, StreamOutput::writeException);
+        out.writeMap(details, (o, v) -> v.writeTo(o));
```

Comment:
Lots of places like this could use `StreamOutput::writeWriteable` instead of creating endless identical lambdas. Maybe worth a follow-up rather than fixing them all here.

Reply:
Yea that's a good idea, I keep forgetting we have this one, I'll try some automated follow-up if I can find one :)